drivers/sensor/bosch/bmp180/bmp180.c

```diff
@@ -0,0 +1,487 @@
+/* Bosch BMP180 pressure sensor
+ *
+ * Copyright (c) 2024 , Chris Ruehl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.mouser.hk/datasheet/2/783/BST-BMP180-DS000-1509579.pdf
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/pm/device.h>
+#include <math.h>
+
+#include "bmp180.h"
+
+LOG_MODULE_REGISTER(BMP180, CONFIG_SENSOR_LOG_LEVEL);
+
+static inline int bmp180_bus_check(const struct device *dev)
+{
+	const struct bmp180_config *cfg = dev->config;
+
+	return cfg->bus_io->check(&cfg->bus);
+}
+
+static inline int bmp180_reg_read(const struct device *dev,
+				  uint8_t start, uint8_t *buf, int size)
+{
+	const struct bmp180_config *cfg = dev->config;
+
+	return cfg->bus_io->read(&cfg->bus, start, buf, size);
+}
+
+static inline int bmp180_reg_write(const struct device *dev, uint8_t reg,
+				   uint8_t val)
+{
+	const struct bmp180_config *cfg = dev->config;
+
+	return cfg->bus_io->write(&cfg->bus, reg, val);
+}
+
+int bmp180_reg_field_update(const struct device *dev,
+			    uint8_t reg,
+			    uint8_t mask,
+			    uint8_t val)
+{
+	int rc = 0;
+	uint8_t old_value, new_value;
+	const struct bmp180_config *cfg = dev->config;
+
+	rc = cfg->bus_io->read(&cfg->bus, reg, &old_value, 1);
+	if (rc != 0) {
+		return rc;
+	}
+
+	new_value = (old_value & ~mask) | (val & mask);
+	if (new_value == old_value) {
+		return 0;
+	}
+
+	return cfg->bus_io->write(&cfg->bus, reg, new_value);
+}
+
+#ifdef CONFIG_BMP180_OSR_RUNTIME
+static int bmp180_attr_set_oversampling(const struct device *dev,
+					enum sensor_channel chan,
+					uint16_t val)
+{
+	struct bmp180_data *data = dev->data;
+
+	/* Value must be a positive value 0-3 */
+	if(val > 3) {
+		return -EINVAL;
+	}
+
+	if (chan == SENSOR_CHAN_PRESS) {
+	     data->osr_pressure = val;
```

Comment:
I would handle the invalid value in the if block and have the normal execution on the main path.

Reply:
Its the think positive approach - 
yeah ,, after looking into it again just make and simplify to this:
if ((chan==SENSOR_CHAN_PRESS) && (val<=3)) {
 // ok
}
else {
 return -EINVAL;
}
thanks to point out your opinion

Reply:
got it. applied.